### ISSUE TITLE
Upgrading to MSBuild 15.8 causes incorrect TargetFramework property in ProjectReference builds

### ISSUE THREAD

**Issue by natemcmaster** [Microsoft/msbuild#3626]
### Steps to reproduce
You'll need VS 2017 with C++ tools to do this.
```
git clone https://github.com/aspnet/IISIntegration --branch namc/repro-netsdk1005
build.cmd /bl /p:SkipTests=true
```

### Expected  behavior
This project builds just fine with MSBuild 15.7. It should continue to do so in 15.8.

Build log for compiling this project with the 2.1.302 SDK: 
[2_1_302.binlog.zip](https://github.com/Microsoft/msbuild/files/2295693/2_1_302.binlog.zip)

### Actual behavior
The project fails to build

> C:\b\w\1973e9914e89f587.dotnet\x64\sdk\2.1.400\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(198,5): error NETSDK1005: Assets file 'C:\b\w\1973e9914e89f587\modules\IISIntegration\src\Microsoft.AspNetCore.Server.IIS\obj\project.assets.json' doesn't have a target for '.NETCoreApp,Version=v2.2'. Ensure that restore has run and that you have included 'netcoreapp2.2' in the TargetFrameworks for your project.

Build log for compiling this project with the 2.1.400 SDK: 
[2_1_400.binlog.zip](https://github.com/Microsoft/msbuild/files/2295685/2_1_400.binlog.zip)

Some how a build of Microsoft.AspNetCore.Server.IIS.csproj is getting triggered with "TargetFramework=netcoreapp2.2" even though [this project only has `<TargetFramework>netstandard2.0</TargetFramework>`](https://github.com/aspnet/IISIntegration/blob/6440545c4a1a170de19ca80c1cf24ff346a005b1/src/Microsoft.AspNetCore.Server.IIS/Microsoft.AspNetCore.Server.IIS.csproj#L4).

![image](https://user-images.githubusercontent.com/2696087/44243308-4b3ee180-a183-11e8-8da9-a9ab4367304b.png)

### Environment data
`dotnet msbuild -version`
```
Microsoft (R) Build Engine version 15.8.166+gd4e8d81a88 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

15.8.166.59604
```
OS info: Windows 10

If applicable, version of the tool that invokes MSBuild (Visual Studio, dotnet CLI, etc): dotnet cli 2.1.400

cc @rainersigwald @dsplaisted @nguerrera

**Comment by nguerrera** [comment]
@natemcmaster I get this before getting that far:

```
C:\Users\nicholg\.dotnet\buildtools\korebuild\2.2.0-preview1-20180816.6\modules\KoreBuild.Tasks\module.targets(154,5): error : Could 
not find an installation of Visual Studio that satisifies the specified requirements in D:\Temp\nate\IISIntegration/korebuild.jsonExecute run.ps1 
install vs to update or install the current vs installation. [C:\Users\nicholg\.dotnet\buildtools\korebuild\2.2.0-preview1-20180816.6\KoreBuild.proj]
```

**Comment by nguerrera** [comment]
I have C++ workload in all VS 2017 instances on my box.

**Comment by nguerrera** [comment]
From binlog and looking at sln, it appears to have something to do with a solution build order dependency specifed from StressTestWebSite -> Microsoft.AspNetCore.Server.IIS. This seems to generate a projectreference at a point in time that does not get the TFM negotiation. I don't know what changed around this.

![image](https://user-images.githubusercontent.com/75470/44244013-ade5ac80-a186-11e8-8ed6-2fefa114c15d.png)

**Comment by nguerrera** [comment]
Possible related change: https://github.com/Microsoft/msbuild/pull/3315

**Comment by dasMulli** [comment]
Here's a minimal repro of what I think is going on (haven't looked at the binlog above): 
[tf-repro.zip](https://github.com/Microsoft/msbuild/files/2296033/tf-repro.zip)

When some project that is built with a `TargetFramework` global property has a solution dependency, the property is now passed along the synthetic p2p reference since there is no re-negoatiation of a compatible TF value.

**Comment by rainersigwald** [comment]
@dasMulli's analysis sounds right, which means that a workaround is available (as in https://github.com/Microsoft/msbuild/issues/2661#issuecomment-338808156).

For the repro project it can be applied as:

```diff
diff --git a/tf-repro/LibMultiTargeting/LibMultiTargeting.csproj b/tf-repro/LibMultiTargeting/LibMultiTargeting.csproj
index f674eb3..3318e06 100644
--- a/tf-repro/LibMultiTargeting/LibMultiTargeting.csproj
+++ b/tf-repro/LibMultiTargeting/LibMultiTargeting.csproj
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
     <TargetFrameworks>netcoreapp2.1;netstandard2.0</TargetFrameworks>
+    <AddSyntheticProjectReferencesForSolutionDependencies>false</AddSyntheticProjectReferencesForSolutionDependencies>
   </PropertyGroup>
 
 </Project>
```

The property can also be specified in a `Directory.Build.props`, and should not cause problems with earlier versions of MSBuild.

**Comment by natemcmaster** [comment]
Thanks for taking a look so quickly. Can confirm that setting AddSyntheticProjectReferencesForSolutionDependencies=false works for my particular project. I'll add this so we can proceed with upgrading to the latest MSBuild.

**Comment by rainersigwald** [comment]
Fixed with #3628 for 15.9.
